Neos.Fusion/Classes/View/FusionView.php

```diff
@@ -179,11 +187,9 @@ protected function loadFusion()
     }
 
     /**
-     * Parse all the fusion files the are in the current fusionPathPatterns
-     *
-     * @return array
+     * Parse all the fusion files that are in the current fusionPathPatterns
      */
-    protected function getMergedFusionObjectTree(): array
+    protected function getMergedFusionObjectTree(): FusionConfiguration
```

Comment:
Is that signature change really needed here? This is used by monocle and would require a new release for Neos 9.

Reply:
Good catch. Was just cleaning up. The alternative from using the DTO through all layers is 

Parser -> DTO -> to array -> to DTO -> Runtime

Reply:
It seems the method is fine as is, bun changing the property type breaks?

https://github.com/sitegeist/Sitegeist.Monocle/blob/5b33c359f88251d9563de1a068d78c32daf87bfe/Classes/Fusion/FusionView.php#L49

Reply:
Monocle calls this method aswell so the signature change there is also relevant.

Reply:
Sorry I can’t find it ^^

Reply:
You are right. Monocle calls its own method getMergedFusionObjectTreeForSitePackage but this had to change signature since parsedFusion changed the type.

Am not strictly against changing this but we already declared monocle Neos 9 compliant some time ago since we expected no changes in this regard as we adjusted the fusion parser stuff in 8.3 for that. 

It would be really great to be able to support Neos 8 and 9 from a single package version since i expect Neos 8 to be still relevant for a while.

Reply:
Hmm i value your point and will not destroy your plan (be the reason for the extra hassle of a dedicated 90 version).
How about adding a tiny little bit of conditional code like:

```php
public function loadFusion() {
    if (FLOW_VERSION === '90') { // or a try catch, or actually asking the package manager
        $this->parsedFusion = $someService->getMergedFusionObjectTreeForSitePackage();
    } else {
        $this->parsedFusion = $someService->getMergedFusionObjectTreeForSitePackage()->toArray();
    }
}
```

i do think its important to add some type safety to help people extending the view point in the right direction and prevent them from foot-guns.

Reply:
Hmmm will try that here and use a custom FusionService:

```
public function loadFusion() {
        $r = new \ReflectionClass(static::class);
        $parsedFusionType = $r->getProperty('parsedFusion')->getType();
        if ($parsedFusionType->getName() === FusionConfiguration::class) {
            $this->parsedFusion = $this->fusionService->getMergedFusionObjectTreeForSitePackage($this->getOption('packageKey'));
        } else {
            $this->parsedFusion = $this->fusionService->getMergedFusionObjectTreeForSitePackage($this->getOption('packageKey'))->toArray();
        }
}
```
Will tell you the result shortly

Reply:
Seems work reasonably well here https://github.com/sitegeist/Sitegeist.Monocle/pull/221 ... will try in 8.3 but that would be fine for me

Reply:
Ok i could adjust monocle to run on 8.3 and 9.0 so i consider this solved.